src/Middleware/LanguageSwitcherMiddleware.php

```diff
@@ -117,11 +117,11 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
         if (!$locale) {
             return $this->__next($request, $response, $next);
         }
-
-        if (in_array($locale, $this->__getAllowedLanguages()) || $this->__getAllowedLanguages() === ['*']) {
+        if ($this->__getAllowedLanguages() !== ['*']) {
+            $locale = Locale::lookup($this->__getAllowedLanguages(), $locale, true, Configure::read('App.defaultLocale'));
```

Comment:
if `__getAllowedLanguages()` returns an empty array, `Locale::lookup()` will NOT return the default locale but an empty string.

Might be worth checking for that?